pkg/cmd/operator/sidecar.go

```diff
@@ -209,142 +198,21 @@ func (o *SidecarOptions) Run(streams genericclioptions.IOStreams, cmd *cobra.Com
 	if err != nil {
-		return fmt.Errorf("can't wait for pod's ContainerID: %w", err)
+		return fmt.Errorf("can't get pod %q: %w", o.ServiceName, err)
 	}
 
 	member, err := identity.NewMember(service, pod, o.nodesBroadcastAddressType, o.clientsBroadcastAddressType)
 	if err != nil {
 		return fmt.Errorf("can't create new member from objects: %w", err)
 	}
 
-	labelSelector := labels.Set{
-		naming.OwnerUIDLabel:      string(pod.UID),
-		naming.ConfigMapTypeLabel: string(naming.NodeConfigDataConfigMapType),
-	}.AsSelector()
-	podLW := &cache.ListWatch{
-		ListFunc: helpers.UncachedListFunc(func(options metav1.ListOptions) (runtime.Object, error) {
-			options.LabelSelector = labelSelector.String()
-			return o.kubeClient.CoreV1().ConfigMaps(pod.Namespace).List(ctx, options)
-		}),
-		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-			options.LabelSelector = labelSelector.String()
-			return o.kubeClient.CoreV1().ConfigMaps(pod.Namespace).Watch(ctx, options)
-		},
-	}
-	klog.V(2).InfoS("Waiting for NodeConfig's data ConfigMap ", "Selector", labelSelector.String())
-	_, err = watchtools.UntilWithSync(
-		ctx,
-		podLW,
-		&corev1.ConfigMap{},
-		nil,
-		func(e watch.Event) (bool, error) {
-			switch t := e.Type; t {
-			case watch.Added, watch.Modified:
-				cm := e.Object.(*corev1.ConfigMap)
-
-				if cm.Data == nil {
-					klog.V(4).InfoS("ConfigMap missing data", "ConfigMap", klog.KObj(cm))
-					return false, nil
-				}
-
-				srcData, found := cm.Data[naming.ScyllaRuntimeConfigKey]
-				if !found {
-					klog.V(4).InfoS("ConfigMap is missing key", "ConfigMap", klog.KObj(cm), "Key", naming.ScyllaRuntimeConfigKey)
-					return false, nil
-				}
-
-				src := &internalapi.SidecarRuntimeConfig{}
-				err = json.Unmarshal([]byte(srcData), src)
-				if err != nil {
-					klog.V(4).ErrorS(err, "Can't unmarshal scylla runtime config", "ConfigMap", klog.KObj(cm), "Key", naming.ScyllaRuntimeConfigKey)
-					return false, nil
-				}
-
-				if src.ContainerID != containerID {
-					klog.V(4).InfoS("Scylla runtime config is not yet updated with our container id",
-						"ConfigMap", klog.KObj(cm),
-						"ConfigContainerID", src.ContainerID,
-						"SidecarContainerID", containerID,
-					)
-					return false, nil
-				}
-
-				if len(src.BlockingNodeConfigs) > 0 {
-					klog.V(4).InfoS("Waiting on NodeConfig(s)",
-						"ConfigMap", klog.KObj(cm),
-						"ContainerID", containerID,
-						"NodeConfig", src.BlockingNodeConfigs,
-					)
-					return false, nil
-				}
-
-				klog.V(4).InfoS("ConfigMap container ready", "ConfigMap", klog.KObj(cm), "ContainerID", containerID)
-				return true, nil
-
-			case watch.Error:
-				return true, apierrors.FromObject(e.Object)
-
-			default:
-				return true, fmt.Errorf("unexpected event type %v", t)
-			}
-		},
-	)
-	if err != nil {
-		return fmt.Errorf("can't wait for optimization: %w", err)
-	}
-
 	klog.V(2).InfoS("Starting scylla")
```

Comment:
the PR description says `[ignition command] signals ScyllaDB container by creating a file that unblocks its start up` - where does that actually happen? I can't seem to find this piece of code. To me it looks like sidecar doesn't wait for anything now?

Reply:
It's been a few months, not sure where it got lost. Added the file wait into the resource defs. Thanks!